src/main/software/amazon/event/ruler/ByteMachine.java

```diff
@@ -648,28 +648,26 @@ private boolean attemptAddShortcutTransitionMatch(final ByteTransition transitio
         return false;
     }
 
-    NameState addPattern(final Patterns pattern) {
-        return addPattern(pattern, null);
-    }
-
     /**
      * Adds one pattern to a byte machine.
      *
      * @param pattern The pattern to add.
      * @param nameState If non-null, transition to this NameState from the ByteMatch.
-     * @return NameState transitioned to from ByteMatch. Will be equal to provided NameState if it wasn't null.
      */
-    NameState addPattern(final Patterns pattern, final NameState nameState) {
+    void addPattern(final Patterns pattern, final NameState nameState) {
```

Comment:
I actually thought about this differently. I wanted to firm up and simplify the contract of ByteMachine.addPattern. Since it will always use the provided NameState, I didn't think it made sense for the interface to return a NameState. I wanted consumers of the class to understand that their provided NameState will always be used. If you're concerned about bugs, we could change access so only GenericMachine can create NameStates (without a workaround), but I feel this might be overkill.